src/main/java/me/rigamortis/seppuku/impl/module/misc/HotBarRefillModule.java

```diff
@@ -0,0 +1,185 @@
+    }
+
+    /**
+     * Returns a given player's hotbar.
+     *
+     * @param player The player who's hotbar should be returned.
+     * @return The passed player's hotbar as a primitive array.
+     */
+    private ItemStack[] getHotbar (EntityPlayerSP player) {
+        ItemStack[] ret = new ItemStack[9];
+
+        if (player != null) {
+            for (int i = 0; i <= 8; i++) {
+                    ret[i] = player.inventory.mainInventory.get(i);
+            }
+        }
+
+        return ret;
+    }
+
+    /**
+     * Retrieves the largest stack of a given ItemStack's Item in the player's inventory.
+     *
+     * @param player The player
+     * @param item The item type that should be found
+     * @param includeHotbar If the search should include the hobar
+     * @return The index of the largest stack of the given item, -1 if the given item does not exist.
+     */
+    public int getBiggestStack (EntityPlayerSP player, ItemStack item, boolean includeHotbar) {
+        if (item == null) {
+            return -1;
+        }
+
+        int maxCount = -1;
+        int maxIndex = -1;
+        for (int i = includeHotbar ? 0 : 9; i < player.inventory.mainInventory.size(); i++) {
+            ItemStack stack = player.inventory.mainInventory.get(i);
+
+            if (stack.getItem() != Items.AIR
+                && stack.getItem() == item.getItem()
+                && stack.getCount() > maxCount) {
+
+                maxCount = stack.getCount();
+                maxIndex = i;
+            }
+        }
+
+        return maxIndex;
+    }
+
+    /**
+     * Refills a given slot in the hotbar from an item in the player's inventory.
+     * Uses the slot's current ItemStack to decide what it should be refilled with.
+     *
+     * @param mc The passed Mincraft instance
+     * @param slot The slot that should be refilled
+     */
+    public void refillHotbarSlot(Minecraft mc, int slot) {
+        ItemStack stack;
+        if (slot == 45) { // Special case for offhand
+            stack = mc.player.getHeldItemOffhand();
+        } else {
+            stack = mc.player.inventory.mainInventory.get(slot);
+        }
+
+        // If the slot is air it cant be refilled
+        if (stack.getItem() == Items.AIR) {
+            return;
+        }
+
+        // The slot can't be refilled if there is nothing to refill it with
+        int biggestStack = getBiggestStack(mc.player, stack, false);
+        if (biggestStack == -1) {
+            return;
+        }
+
+        // Special case for offhand (can't use QUICK_CLICK)
+        if (slot == 45) {
+            mc.playerController.windowClick(mc.player.inventoryContainer.windowId, biggestStack, 0, ClickType.PICKUP, mc.player);
+            mc.playerController.windowClick(mc.player.inventoryContainer.windowId, 45, 0, ClickType.PICKUP, mc.player);
+            mc.playerController.windowClick(mc.player.inventoryContainer.windowId, biggestStack, 0, ClickType.PICKUP, mc.player);
+            return;
```

Comment:
Do we know why this is a special case and why QUICK_CLICK doesn't work?  I'm personally curious about it.

Reply:
Only shields can be shift-clicked into the offhand, since thats what the offhand was originally made for.